File: fe/src/types/employee.ts
export interface Employee {
    id: string;
    name: string;
    email: string;
    phone: string;
    position: string;
    department: string;
    salary: number;
    address: string;
    status: string;
    created_at: string;
    updated_at: string;
    deleted_at: string | null;
}

export const positionLabel: Record<number, string> = {
    2: "HR",
    3: "Admin",
    4: "Employee"
};

export const departmentLabel: Record<number, string> = {
    1: "HRD",
    2: "Engineering / IT",
    3: "Keuangan",
    4: "Operasional",
    5: "Sales & Marketing",
};

export const statusLabel: Record<number, string> = {
    2: "Aktif",
    1: "Tidak Aktif",
};

export const optionPosition = [
    { value: "2", label: "HR" },
    { value: "3", label: "Admin" },
    { value: "4", label: "Employee" },
];

export const optionDepartment = [
    { value: "1", label: "HRD" },
    { value: "2", label: "Engineering / IT" },
    { value: "3", label: "Keuangan" },
    { value: "4", label: "Operasional" },
    { value: "5", label: "Sales & Marketing" },
];

export const optionStatus = [
    { value: "1", label: "Aktif" },
    { value: "2", label: "Tidak Aktif" },
];

File: domain/entity/employee.go
package entity

import (
	"database/sql"
	"time"

	"github.com/google/uuid"
	"github.com/mqqff/absensi-app/domain/enums"
)

type Employee struct {
	ID         uuid.UUID                   `db:"id" json:"id"`
	Name       string                      `db:"name" json:"name"`
	Email      string                      `db:"email" json:"email"`
	Phone      string                      `db:"phone" json:"phone"`
	Position   enums.EmployeePositionIdx   `db:"position" json:"position"`
	Department enums.EmployeeDepartmentIdx `db:"department" json:"department"`
	Salary     float64                     `db:"salary" json:"salary"`
	Address    string                      `db:"address" json:"address"`
	Password   string                      `db:"password" json:"-"`
	Status     enums.EmployeeStatus        `db:"status" json:"status"`
	CreatedAt  time.Time                   `db:"created_at" json:"created_at"`
	UpdatedAt  time.Time                   `db:"updated_at" json:"updated_at"`
	DeletedAt  sql.NullTime                `db:"deleted_at" json:"deleted_at"`
}
